python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -8429,3 +8429,36 @@ def test_rename_for_level_is_None_MC():
     got = gdf.rename(columns={"a": "f"}, level=None)
 
     assert_eq(expect, got)
+
+
+@pytest.mark.parametrize("ignore_index", [True, False])
+@pytest.mark.parametrize("p_index", [None, ['ia', 'ib', 'ic']])
+def test_explode(ignore_index, p_index):
+    gdf = cudf.DataFrame({
+        "a": [[1, 2, 3], [4, 5], [6]],
+        "b": [11, 22, 33],
+        "c": [111, 222, 333]
+    }, index=p_index)
+    pdf = gdf.to_pandas()
+
+    expect = pdf.explode('a', ignore_index)
+    got = gdf.explode('a', ignore_index)
+
+    assert_eq(expect, got, check_dtype=False)
+
+
+@pytest.mark.xfail(
+    reason="nulls are dropped by cudf, but pandas casts it to NaN"
+)
+def test_explode_with_nulls():
+    gdf = cudf.DataFrame({
+        "a": [[1, 2, 3], [4, 5], None],
+        "b": [11, 22, 33],
+        "c": [111, 222, 333]
+    })
```

Comment:
We can't silently return different results than Pandas here. We need to wait for `explode_outer` support in libcudf and use that instead.

Also need to cover empty lists here as well.

Reply:
Thanks for reivew. will draft this to wait for `explode_outer ` support in libcudf.